### PR TITLE
Add preauth-simulation-filter-servlet_2.5 module, which is essentially a...

### PR DESCRIPTION
**Pull request by ctcudd**
... backport of preauth-simulation-filter that works with jdk 6 and servlet 2.5.

### CODE REVIEW

File: preauth-simulation-filter-servlet_2.5/pom.xml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	<description>Servlet Filter useful for simulating Pre-Authentication environment in a servlet 2.5 environment. This allows a developer to spoof request headers or environment variables that would provided by preauth tools like Shibboleth and Siteminder without having those installed locally.</description>
+	
+	<dependencies>
+		<dependency>
+			<groupId>javax.servlet</groupId>
+			<artifactId>servlet-api</artifactId>
+			<version>2.5</version>
+		</dependency>
+		<dependency>
+			<groupId>org.apache.commons</groupId>
```
Comment (nblair) [inline preauth-simulation-filter-servlet_2.5/pom.xml]: Could we drop the commons-lang3 dependency? It looks like it's used only on one line in PreAuthenticationSimulationServletFilter, i'd rather not have these modules pull in extra dependencies.

**Comment by nblair** [comment]
The top comment suggest this targets JDK 6, but there is nothing in the pom that actually enforces that. I'd expect to see a reference to maven-compiler-plugin reference in the 2.5 module that sets 1.6 for source and target.

Currently, the configuration for maven-compiler-plugin in the parent pom takes precedence, which is configured to use 1.7 for source and target. Will your environment support 1.7 target jars? Or do you specifically need support for 1.6?

**Comment by nblair** [comment]
If you don't specifically need to run on JDK 6, I can go ahead and merge this; if you do need JDK 6 then we need to brainstorm a bit.

**Comment by ctcudd** [comment]
I DO specifically need jdk6, target and source are set to 1.6 in the pom, see:  https://github.com/ctcudd/developer-tools/blob/c3bcfb738b35e2fa144b892fea6c8679e767c055/preauth-simulation-filter-servlet_2.5/pom.xml

**Comment by ctcudd** [comment]
ctcudd:developer-tools ctcudd$ javap -verbose preauth-simulation-filter/target/classes/com/github/nblair/web/PreAuthenticationSimulationServletFilter.class  | grep "major\|minor"
  minor version: 0
  major version: 51

ctcudd:developer-tools ctcudd$ javap -verbose preauth-simulation-filter-servlet_2.5/target/classes/com/github/nblair/web/PreAuthenticationSimulationServletFilter.class  | grep "major\|minor"
  minor version: 0
  major version: 50

**Comment by nblair** [comment]
Ah ha - I missed that. Looks good!

I'll ship an 0.2.0 release shortly.

**Pull request merged**